src/DeviceInterface.cpp

```diff
@@ -46,6 +47,32 @@ bool lookup_runtime_routine(const char *name, const Target &target,
 
 }
 
+namespace Halide {
+halide_device_interface_t *get_device_interface_for_target(const Target &t) {
```

Comment:
Either passing just a feature flag or using DeviceAPI instead of a Target seems safer here.